src/zhinst/toolkit/nodetree/nodetree.py

```diff
@@ -489,6 +493,12 @@ def string_to_raw_path(self, node: str) -> str:
                 ) from error
         return node.lower()
 
+    def send_or_add2transaction(self, settings: t.List[t.Tuple[t.Union[Node, str], t.Any]]):
+        if self.transaction.in_progress():
+            self.transaction.add_list(settings)
+        else:
+            self._session.daq_server.set(settings)
+
```

Comment:
Yes it is always good to be as versatile as possible. But on this level performance is more important 😬 .

I would therefor not create a function that accepts both strings and Node objects but only what you actually need. This means a function that takes a list of string(raw nodes) and values pairs and directly adds this to `self._queue`. 

Also, the name can be simplified. The class is already called Transaction so no need to add it in the function name. I would go with something like `send_or_append_raw`.